### ISSUE TITLE
MIDI map does not save on Windows 10

### ISSUE THREAD

**Issue by megalon** [cpyarger/obs-midi#182]
## Platform
Operating system and version: 
Windows 10 Version 10.0.19044 Build 19044

OBS Studio version:
27.2.4

## Expected Behavior
When I create a MIDI map through the `Tools > Config > OBS Midi Settings`, it should load this MIDI map the next time I launch OBS

## Current Behavior
When I close and relaunch OBS, the MIDI map is blank. 
I also don't have any files in my `obs-studio\plugin_config\obs-midi` folder. It is completely empty.

## Steps to Reproduce
1. Launch OBS
1. Navigate to `Tools > Config > OBS Midi Settings`
1. Map a few controls to my MIDI controller
1. Test that the controls work correctly
1. Close the config window
1. Close OBS
1. Relaunch OBS
1. None of the mapped controls work anymore, and when I open the `Tools > Config > OBS Midi Settings` menu, the list of mapped controls is empty.
![image](https://user-images.githubusercontent.com/27714637/161138125-e41e763f-d8b9-4240-a05b-336f9ce48288.png)

## Additional information
I am using a LIVID OhmRGB 64 MIDI controller. It works fine, and the MIDI map does work correctly, as long as I don't close OBS after I create it.
It seems like the config files just aren't saving.
Log file: [2022-03-19 18-43-47.txt](https://github.com/cpyarger/obs-midi/files/8392683/2022-03-19.18-43-47.txt)

**Comment by kineticscreen** [comment]
You almost certainly have a scene collection name, or scene name or source name that is too long, or has unusual characters. There is a bug in obs-midi that causes the bindings to be corrupted.

**Comment by megalon** [comment]
Thanks, that sounds frustrating to troubleshoot.

I ended up using OSC-for-OBS instead of obs-midi. I use a custom program written in Pure Data to interface with it.
<https://github.com/jshea2/OSC-for-OBS>

For reference though, my longest source name is about 30 characters.
And the only characters I use in my source / scene names are alpha numeric, space, and "-"

**Comment by kineticscreen** [comment]
Yeah that 30 character name would do it. Even the dash could be a problem. I had repeated issues on a recent show and just made everything incredibly short just to be safe, so I haven't worked out what the threshold is. 

EDIT: oh wow, OSC for OBS sounds great. And I was integrating with Qlab so sounds like a much better solution.

**Comment by InfoSec812** [comment]
I am seeing this intermittently on Linux as well.

**Comment by gitagogaming** [comment]
This has happened to me time and time again.. its beyond frustrating.. 
I most certainly use - in my collection names

**Comment by cpyarger** [comment]
OBS recently updated to QT6 and so I will be doing another rewrite of things soon.  My apologies for things taking so long, I currently am the only one working on obs-midi, and I just don't have the time or money to focus on it full time like I have in the past. If anyone wants to help out I would gladly appreciate contributors 😸

**Comment by InfoSec812** [comment]
@cpyarger I will try to take a look in the near future to see if I can isolate where the issue is occurring. I think I would like to see at the same time if there could be a simple path to add an "export" button as well. I will likely be early September before I have time though. Thanks again for all of you contributions here!
